src/utils/getstream/useChatClientHook.js

```diff
@@ -1,12 +1,12 @@
 import {putUserTopic} from '../../service/topics';
 
 const useChatClientHook = () => {
-  const followTopic = async (topic) => {
+  const followTopic = async (topic, isIncognito) => {
     const data = {
       name: topic
     };
 
-    const response = await putUserTopic(data);
+    const response = await putUserTopic(data, isIncognito);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `followTopic` function has been updated to accept an `isIncognito` parameter, which is then passed to the `putUserTopic` function. This change seems to be in line with the PR's goal of supporting anonymous following. However, it would be beneficial to add error handling here to catch any potential issues that might arise during the async operation.

```diff
  const followTopic = async (topic, isIncognito) => {
     const data = {
       name: topic
     };
+    try {
-    const response = await putUserTopic(data, isIncognito);
+      const response = await putUserTopic(data, isIncognito);
+      return response?.data;
+    } catch (error) {
+      console.error('Error following topic:', error);
+      // Consider rethrowing the error or handle it appropriately based on your application's needs
+    }
-    return response?.data;
  };
```
This way, if there's an error while calling `putUserTopic`, it will be caught and logged, preventing the application from crashing due to unhandled promise rejection.


<!-- This is an auto-generated comment by OSS CodeRabbit -->